compiler/rustc_typeck/src/check/pat.rs

```diff
@@ -1040,6 +1042,33 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
                 vec![(left, "(".to_string()), (right.shrink_to_hi(), ")".to_string())],
                 Applicability::MachineApplicable,
             );
+        } else if fields.len() > subpats.len() {
+            let after_fields_span = if pat_span == DUMMY_SP {
+                pat_span
```

Comment:
What is the `DUMMY_SP` for and this branch for? I never seen this before.

Reply:
This is to prevent a compiler crash if `pat_span == DUMMY_SP` since we then subtract `1` from `pat_span` and `DUMMY_SP` is basically equal to 0.

Reply:
If `pat_span` is `DUMMY_SP` then you shouldn't be giving suggestions at all.

Reply:
What happens if the error rendering system receives a diagnostic that has `DUMMY_SP`? Will it just ignore it?

Reply:
The error renderer will ignore `DUMMY_SP`s and not show labels associated to it, while still showing messages, but they do have the potential to break suggestions and _certainly_ break everything when synthesizing a new span from them.